test/compilable/test22122.d

```diff
@@ -0,0 +1,36 @@
+// EXTRA_FILES: imports/imp22122.d
+module imports.test22122;
+
+struct S22122
+{
+    import imports.imp22122;
+    Variant!(Imp22122)[] array;
+}
+
+void test22122(S22122 s)
+{
+    try
+    {
+        foreach(elem; s.array)
+        {
+            import imports.imp22122;
+            with(elem.get!Imp22122)
+            {
+            }
+        }
+    }
+    finally
+    {
+    }
+}
```

Comment:
Maybe add another tests for `TryCatchStatement` given that the existing tests allowed this regression to slip by for both (try-catch, try-finally)